crates/aptos/src/common/transactions.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::{
+    local_simulation,
+    types::{
+        AccountType, CliConfig, CliError, CliTypedResult, ConfigSearchMode, EncodingOptions,
+        ExtractPublicKey, GasOptions, PrivateKeyInputOptions, ProfileOptions, PromptOptions,
+        RestOptions, TransactionSummary, ACCEPTED_CLOCK_SKEW_US, US_IN_SECS,
+    },
+    utils::{get_account_with_state, get_sequence_number},
+};
+use aptos_api_types::ViewFunction;
+use aptos_crypto::{
+    ed25519::{Ed25519PrivateKey, Ed25519PublicKey},
+    hash::CryptoHash,
+};
+use aptos_move_debugger::aptos_debugger::AptosDebugger;
+use aptos_rest_client::Client;
+use aptos_sdk::{transaction_builder::TransactionFactory, types::LocalAccount};
+use aptos_types::{
+    account_address::AccountAddress,
+    chain_id::ChainId,
+    transaction::{
+        authenticator::{AccountAuthenticator, TransactionAuthenticator},
+        SignedTransaction, TransactionPayload, TransactionStatus,
+    },
+};
+use aptos_vm_types::output::VMOutput;
+use clap::Parser;
+use move_core_types::vm_status::VMStatus;
+pub use move_package::*;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+/// Common options for simulating and running transactions
+///
+/// NOTE: this should be separated out from the existing TransactionOptions, which is now too big
+///
+/// TODO: I know this is a copy of the existing one, but it should streamline a lot of the code.  Currently experimental without any worries of backwards compatibility
+#[derive(Debug, Default, Parser)]
+pub struct TxnOptions {
+    /// Sender account address
+    ///
+    /// This allows you to override the account address from the derived account address
+    /// in the event that the authentication key was rotated or for a resource account
+    #[clap(long, value_parser = crate::common::types::load_account_arg)]
+    pub(crate) sender_account: Option<AccountAddress>,
+
+    #[clap(flatten)]
+    pub(crate) private_key_options: PrivateKeyInputOptions,
+    #[clap(flatten)]
+    pub(crate) encoding_options: EncodingOptions,
+    #[clap(flatten)]
+    pub(crate) profile_options: ProfileOptions,
+    #[clap(flatten)]
+    pub(crate) rest_options: RestOptions,
+    #[clap(flatten)]
+    pub(crate) gas_options: GasOptions,
+    #[clap(flatten)]
+    pub prompt_options: PromptOptions,
+}
+
+impl TxnOptions {
+    /// Builds a rest client
+    fn rest_client(&self) -> CliTypedResult<Client> {
+        self.rest_options.client(&self.profile_options)
+    }
+
+    pub fn get_transaction_account_type(&self) -> CliTypedResult<AccountType> {
```

Comment:
What error do you get if there is no default profile and you don't specify `--private-key` or `--private-key-file`? Ideally not just `One of ['--private-key', '--private-key-file'] or a profile must be used`, it should also mention profile stuff.

Reply:
There is: **or a profile must be used**

Not the best writing / should say `--profile`